openstack/identity/identity.go

```diff
@@ -205,71 +205,68 @@ func validateProjectRole(client *gophercloud.ServiceClient, token string, projec
 	return false
 }
 
-func (h Handler) tenantToken(r *http.Request, tenant string) bool {
-	token := r.Header["X-Auth-Token"]
-	if token == nil {
-		return false
-	}
+func (h Handler) checkToken(r *http.Request, tenant string, token string) bool {
```

Comment:
Even though this function is not exported, it would help maintenance to have a comment preceding it.  You describe the intent in the commit message, which is somewhat complex.